src/main/java/com/hivemq/client/internal/mqtt/handler/publish/outgoing/MqttAckSingleFlowable.java

```diff
@@ -66,6 +66,7 @@ protected void subscribeActual(final @NotNull Subscriber<? super Mqtt5PublishRes
         private static final int STATE_NONE = 0;
         private static final int STATE_RESULT = 1;
         private static final int STATE_REQUESTED = 2;
+        private static final int STATE_CANCELLED = 3;
```

Comment:
Wouldnt it be bether to extract the state constants? As they exist repeatedly.

Reply:
Sometimes they have additional of different states and I do not want to introduce unnecessary coupling as this is internal state